core/trino-spi/src/main/java/io/trino/spi/block/MapHashTables.java

```diff
@@ -166,6 +166,7 @@ synchronized void buildHashTableStrict(Block keyBlock, int keyOffset, int keyCou
         int hashTableSize = keyCount * HASH_MULTIPLIER;
 
         for (int i = 0; i < keyCount; i++) {
+            // this throws if the position is null
```

Comment:
Maybe split out the map and row type changes, since they're more substantial

Reply:
I split out the map change, but the row changes are pretty simple